### PR TITLE
Fix broken ENGINE.md link in v6 upgrade.md

### PR DESCRIPTION
**Pull request by markyrbc**

### CODE REVIEW

File: versioned_docs/version-6.x.x/other-topics/upgrade.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Sequelize.useCLS(namespace);
 
 ### Database Engine Support
 
-We have updated our minimum supported database engine versions. Using older database engine will show `SEQUELIZE0006` deprecation warning. Please check [ENGINE.md](https://github.com/sequelize/sequelize/blob/main/ENGINE.md) for version table.
+We have updated our minimum supported database engine versions. Using older database engine will show `SEQUELIZE0006` deprecation warning. Please check [ENGINE.md](https://github.com/sequelize/sequelize/blob/v6.20.1/ENGINE.md) for version table.
```
Comment (WikiRik) [inline versioned_docs/version-6.x.x/other-topics/upgrade.md:27]: Can we make this the v6 branch?

Comment (ephys) [reply versioned_docs/version-6.x.x/other-topics/upgrade.md:27]: Let's point it to https://sequelize.org/releases/ instead

Comment (markyrbc) [reply versioned_docs/version-6.x.x/other-topics/upgrade.md]: @ephys done, let me know if the wording is OK

**Review by ephys — APPROVED** [review]

File: versioned_docs/version-6.x.x/other-topics/upgrade.md (hunk 1 above)
Comment (ephys) [reply versioned_docs/version-6.x.x/other-topics/upgrade.md]: Looks good to me! :)

**Review by WikiRik — APPROVED** [review]
Thanks for fixing this!

**Comment by WikiRik** [comment]
The CI is failing but that's not due to you

**Pull request merged**